docs/proposals/custom-sidecar-config.md

```diff
@@ -0,0 +1,172 @@
+# Custom Container Configuration
+
+## Introduction
+Users and field team have requested access to custom configuration of kubernetes sidecar and init containers.
+See [#3925](https://github.com/kumahq/kuma/issues/3925).
+This intuitively aligns with the ProxyTemplate configuration,
+which provides a free-form configuration of Envoy.
+The current plan is to introduce a new CRD,
+similar to ProxyTemplate,
+which can modify sidecar config utilizing [jsonpatch](https://jsonpatch.com).
```

Comment:
nit: jsonpatch.com seems a little abandoned maybe link to the rfc directly: https://datatracker.ietf.org/doc/html/rfc6902